clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -458,6 +458,10 @@ LogicalResult CastOp::verify() {
     return success();
   }
   case cir::CastKind::bitcast: {
+    // Allow bitcast of structs for calling conventions.
+    if (isa<StructType>(srcType) || isa<StructType>(resType))
```

Comment:
Feels hacky, I don't see documentation supporting this or a good explanation of the semantics. Sounds like we need a new style of cast operation instead.

Reply:
> Feels hacky, I don't see documentation supporting this or a good explanation of the semantics.

I don't think I follow. I'ts a bitcast. We reinterpret struct bits as integers for the calling convention.

> Sounds like we need a new style of cast operation instead.

Do you have something in mind?